desktop_version/src/Editor.cpp

```diff
@@ -2003,6 +2011,39 @@ static void editormenuactionpress(void)
             break;
         }
         break;
+    case Menu::ed_font:
+    {
+        uint8_t font_idx_8x8 = font::get_font_idx_8x8();
+        int option_cur = 0;
+        uint8_t selected = font::get_font_idx_8x8();
+        for (size_t i = 0; i < loc::languagelist.size(); i++)
+        {
+            if (loc::languagelist[i].font_idx != font_idx_8x8)
+            {
+                if (option_cur == game.currentmenuoption)
+                {
+                    selected = loc::languagelist[i].font_idx;
+                    break;
+                }
+                option_cur++;
+            }
+        }
+        cl.level_font_name = font::get_main_font_name(selected);
```

Comment:
Between this and `case Menu::ed_font` in `Game::createmenu`, I'm having a hard time wrapping my head around the logic here. It seems like the point of `option_cur` is to exclude the 8x8 font from selection? (But then why is there an "other" option? Is that not the 8x8 font? What does "other" even mean?) If so, then wouldn't it be clearer to make an explicit list of fonts somewhere else, and then have this menu simply reference that?

Reply:
The font screen looks like this:

![Font screen](https://user-images.githubusercontent.com/44736680/218188397-c2eeb085-ee4c-4a36-90b2-293eb50abb81.png)

So, you get an option for each language that needs a special font, and then the "other" option makes the most sense at the bottom - it's a "none of the above", because all other languages use the 8x8 font.

Reply:
Okay, so "other" is in fact the 8x8 font. Why does the code seemingly exclude it only to add it again later?

Reply:
So that it's at the bottom of the list ("other language than all of the above"), and not unrightfully tied to "English" (or any of the other ~15 languages specifically).

Reply:
So it's a de-duplication measure. Why are only languages with the 8x8 font de-duplicated and not any other fonts that might be used by multiple languages?

Reply:
Well, I don't really expect any other font than 8x8 to be used by more than one language (the han unification makes CJK characters render mutually incorrect across languages, so you have to have language-specific fonts, and CJK fonts do often target only one of the languages). And even if a font is reused, then you just get two options to choose from that result in the same font (without prioritizing one language and hiding the other, or implementing a system that can display multiple languages in a single option like `繁體中文/한국어`, but that system may also go unused).

In the meantime, there are definitely a lot of languages that use the 8x8 font, so the menu would be filled with language options that all have the same effect.

For what it's worth: Dicey Dungeons also has separate fonts for Simplified Chinese, Traditional Chinese, Korean and Japanese in its files, and there is only one font that is used by more than one language: the Latin+Cyrillic one.

Reply:
I was thinking that if there were multiple languages with the same font, it would confuse the user if they saw a different language being selected that wasn't the one they had selected, even though both use the same font. This is more of a problem if the user adds some fonts of their own. And this is why I don't think a language ⟷ font mapping is perfect and the list should just be based off of the actual fonts themselves instead of the languages, even if most of the names are going to be something like "the 한국어 font".

Reply:
To be clear, the fonts in this menu are only global/main ones, not level-specific custom ones, because the font also applies to the title and description in the levels list. If you want to use a completely custom font in a level, you'd have to use `setfont`. Or just override the assets for the global font in your level by giving them the same name, so that it's the standard global font in the levels list and automatically turns into the custom font in the level.

I have actually considered a "font name" system ("the 한국어 font"), but thought it'd be simpler to just take the names from the language list, since we already had them, it was probably going to be a 1:1 mapping, and I didn't want font names to become anything other than language names. But I guess it wouldn't matter too much if fonts need to have a "name" attribute that also specifies the language name again.

Reply:
Yeah, I'd rather have an explicit list of font names rather than fetching it through the language list (and then asking the user to select the language when in reality they're selecting the font). It would make the code clearer.

While you're at it, please don't call the 8x8 font "other". That gives me vibes of MuseScore 3's "New All" button. Name it anything other than "other", maybe something like "default", "standard", or even "latin/cyrillic alphabet" if you have to.